tests/test_preprocessor.py

```diff
@@ -238,3 +238,34 @@ def test_macro_expansion_hinting_bug(self):
 
             with self.assertRaises(EqlTypeMismatchError):
                 parse_query("foo where SELF(bar) == 'baz'")
+
+    def test_append_attribute(self):
+        """Test that macros can expand attributes."""
+        preprocessor = get_preprocessor("""
+            macro NAME(dictfield)   dictfield.name
+            macro ZERO(arrayfield)  arrayfield[0]
+
+            macro ZNAME(objarray)   objarray[0].name
+            macro NAMEZ(arrayobj)   arrayobj.name[0]
+            macro ZNAME2(objarray)  NAME(ZERO(objarray))
+            macro NAMEZ2(arrayobj)  ZERO(NAME(arrayobj))
+        """)
```

Comment:
❤️ 

Are we covered for this too?

```
macro NAMEZZ(subfield)   alert.subfield.name
```

Reply:
i don't have that one covered. it could have unintentional collisions, so i'm not sure how to avoid it. but i'm open to suggestions